controlplane/internal/controller/openshiftassistedcontrolplane_controller.go

```diff
@@ -492,3 +528,69 @@ func (r *OpenshiftAssistedControlPlaneReconciler) ensurePullSecret(
 	acp.Spec.Config.PullSecretRef = &corev1.LocalObjectReference{Name: secret.Name}
 	return nil
 }
+
+func (r *OpenshiftAssistedControlPlaneReconciler) setOpenshiftVersionInStatus(ctx context.Context, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) error {
+	log := ctrl.LoggerFrom(ctx)
+	if !isKubeconfigAvailable(oacp) {
+		return nil
+	}
+	kubeconfigSecret := r.getKubeconfigSecret(ctx, oacp)
+	if kubeconfigSecret == nil {
+		log.Info("kubeconfig secret not found")
+		return nil
+	}
+	kubeconfig, err := util.ExtractKubeconfigFromSecret(kubeconfigSecret)
+	if err != nil {
+		err = errors.Join(err, fmt.Errorf("failed to extract kubeconfig from secret %s", kubeconfigSecret.Name))
+		return err
+	}
+
+	workloadClient, err := workloadclient.GetWorkloadClusterClient(kubeconfig)
+	if err != nil {
+		err = errors.Join(err, fmt.Errorf("failed to establish client for workload cluster from kubeconfig"))
+		return err
+	}
+
+	var clusterVersion configv1.ClusterVersion
+	err = workloadClient.Get(ctx, types.NamespacedName{Name: "version"}, &clusterVersion)
+	if err != nil {
+		err = errors.Join(err, fmt.Errorf(("failed to get ClusterVersion from workload cluster")))
+		return err
+	}
+
+	oacp.Status.DistributionVersion = clusterVersion.Status.Desired.Version
+	return nil
+}
+
+// isKubeconfigAvailable returns true if the openshift assisted control plane
+// condition KubeconfigAvailable is true
+func isKubeconfigAvailable(oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) bool {
+	kubeconfigFoundCondition := util.FindStatusCondition(oacp.Status.Conditions, controlplanev1alpha2.KubeconfigAvailableCondition)
+	if kubeconfigFoundCondition == nil {
+		return false
+	}
+	return kubeconfigFoundCondition.Status == corev1.ConditionTrue
+}
+
+func (r *OpenshiftAssistedControlPlaneReconciler) getKubeconfigSecret(ctx context.Context, oacp *controlplanev1alpha2.OpenshiftAssistedControlPlane) *corev1.Secret {
+	// Retrieve clusterdeployment
+	cd := &hivev1.ClusterDeployment{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Namespace: oacp.Status.ClusterDeploymentRef.Namespace, Name: oacp.Status.ClusterDeploymentRef.Name}, cd); err != nil {
+		return nil
+	}
+	if cd.Spec.ClusterInstallRef == nil {
+		return nil
+	}
```

Comment:
We already handle this: https://github.com/openshift-assisted/cluster-api-agent/blob/master/controlplane/internal/controller/agentclusterinstall_controller.go#L138-L150

I think we can extract common code, and we would already know what is the expected path for the kubeconfig (the one we copy from ACI's). It will save us 2 GET and simplify the code

Reply:
Great idea! Extracted it out to a util function. What do you think?